### PR TITLE
Port WithAcceptanceTest to TCK

### PR DESCRIPTION
**Pull request by petraselmer**

### CODE REVIEW

**Comment by petraselmer** [comment]
@Mats-SX : please check this file against the original. I had to amend quite a few of the queries to fit in with the new process and want to ensure no subtleties were lost in translation...thanks

File: tck/features/WithAcceptance.feature [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+
+  Scenario: Passing on pattern nodes
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A)-[:REL]->(b:B)
+      """
+    When executing query:
+      """
+      MATCH (a:A) WITH a
```
Comment (Mats-SX) [inline tck/features/WithAcceptance.feature]: `WITH a` on new line

File: tck/features/WithAcceptance.feature [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+# limitations under the License.
+#
+
+Feature: WithAcceptance
+
+  Scenario: Passing on pattern nodes
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A)-[:REL]->(b:B)
```
Comment (Mats-SX) [inline tck/features/WithAcceptance.feature]: No need to name `a` and `b`.

File: tck/features/WithAcceptance.feature [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+    Then the result should be:
+      | a    | b    |
+      | (:A) | (:B) |
+    And no side effects
+
+  Scenario: ORDER BY and LIMIT can be used
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A)-[:REL]->(), (b), (c)
```
Comment (Mats-SX) [inline tck/features/WithAcceptance.feature]: We didn't yet decide anything on this, but I prefer:

```
CREATE (a:A), (b), (), ()
CREATE (a)-[:REL]->(b)
```

File: tck/features/WithAcceptance.feature [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+    Then the result should be:
+      | a    |
+      | (:A) |
+    And no side effects
+
+  Scenario: No dependencies between the query parts
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A), (b:B)
```
Comment (Mats-SX) [inline tck/features/WithAcceptance.feature]: May be anonymous.

File: tck/features/WithAcceptance.feature [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+      | (:B) | (:B) |
+    And no side effects
+
+  Scenario: Aliasing
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:Begin {prop: 42}),
+        (b:End {prop: 42}),
+        (c:End {prop: 3})
```
Comment (Mats-SX) [inline tck/features/WithAcceptance.feature]: May be anonoymous

File: tck/features/WithAcceptance.feature [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+      | (:End {prop: 42}) |
+    And no side effects
+
+  Scenario: Handle dependencies across WITH
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:End {prop: 42}),
+        (b:End {prop: 3}),
+        (c:Begin {prop: id(a)})
```
Comment (Mats-SX) [inline tck/features/WithAcceptance.feature]: `b` and `c` may be anonymous

File: tck/features/WithAcceptance.feature [hunk 7]
```diff
@@ -0,0 +1,361 @@
...
+      | (:End {prop: 42}) |
+    And no side effects
+
+  Scenario: Handle dependencies across WITH with SKIP
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a {prop: 'A', id: 0}),
+          (b {prop: 'B', id: id(a)}),
+          (c {prop: 'C', id: 0})
```
Comment (Mats-SX) [inline tck/features/WithAcceptance.feature]: `b` and `c` may be anonymous

File: tck/features/WithAcceptance.feature [hunk 8]
```diff
@@ -0,0 +1,361 @@
...
+      | ({prop: 'A', id: 0}) |
+    And no side effects
+
+  Scenario: WHERE after WITH should filter results
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a {name: 'A'}),
+        (b {name: 'B'}),
+        (c {name: 'C'})
```
Comment (Mats-SX) [inline tck/features/WithAcceptance.feature]: All may be anonymous

File: tck/features/WithAcceptance.feature [hunk 9]
```diff
@@ -0,0 +1,361 @@
...
+  Scenario: WHERE after WITH can filter on top of an aggregation
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a {name: 'A'}),
+        (b {name: 'B'})
+      CREATE (a)-[:REL]->()
+      CREATE (a)-[:REL]->()
+      CREATE (a)-[:REL]->()
+      CREATE (b)-[:REL]->()
```
Comment (Mats-SX) [inline tck/features/WithAcceptance.feature]: As above, we didn't decide on this, but I prefer using a new `CREATE` to distinguish between creating nodes and creating relationships, and to not repeat it if the pattern is very similar:

```
CREATE (a {name: 'A'}), (b {name: 'B'})
CREATE (a)-[:REL]->(),
       (a)-[:REL]->(),
       (a)-[:REL]->(),
       (b)-[:REL]->()
```

or line break with vertical alignment:

```
CREATE (a {name: 'A'}), 
       (b {name: 'B'})
CREATE (a)-[:REL]->(),
       (a)-[:REL]->(),
       (a)-[:REL]->(),
       (b)-[:REL]->()
```

File: tck/features/WithAcceptance.feature [hunk 10]
```diff
@@ -0,0 +1,361 @@
...
+      | ({name: 'A'}) |
+    And no side effects
+
+  Scenario: ORDER BY on an aggregating key
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a1 {bar: 'A'}),
+        (a2 {bar: 'A'}),
+        (b {bar: 'B'})
```
Comment (Mats-SX) [inline tck/features/WithAcceptance.feature]: All may be anonymous

File: tck/features/WithAcceptance.feature [hunk 11]
```diff
@@ -0,0 +1,361 @@
...
+      | 'B'  | 1        |
+    And no side effects
+
+  Scenario: ORDER BY a DISTINCT column
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a1 {bar: 'A'}),
+        (a2 {bar: 'A'}),
+        (b {bar: 'B'})
```
Comment (Mats-SX) [inline tck/features/WithAcceptance.feature]: All may be anonymous

File: tck/features/WithAcceptance.feature [hunk 12]
```diff
@@ -0,0 +1,361 @@
...
+      | 'B'  |
+    And no side effects
+
+  Scenario: WHERE on a DISTINCT column
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a1 {bar: 'A'}),
+        (a2 {bar: 'A'}),
+        (b {bar: 'B'})
```
Comment (Mats-SX) [inline tck/features/WithAcceptance.feature]: All may be anonymous

File: tck/features/WithAcceptance.feature [hunk 13]
```diff
@@ -0,0 +1,361 @@
...
+    Then the result should be:
+      | bars |
+      | 'B'  |
+    And no side effects
+
+  Scenario: A simple pattern with one bound endpoint
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A)-[:REL]->(b:B)
```
Comment (Mats-SX) [inline tck/features/WithAcceptance.feature]: `a` and `b` may be anonymous

File: tck/features/WithAcceptance.feature [hunk 14]
```diff
@@ -0,0 +1,361 @@
...
+    Given an empty graph
+    When executing query:
+      """
+      OPTIONAL MATCH (a:Start)
+      WITH a
+      MATCH (a)-->(b)
+      RETURN *
+      """
+    Then the result should be:
+      | a | b | r |
```
Comment (Mats-SX) [inline tck/features/WithAcceptance.feature]: This one is interesting. I've always used `Then the result should be empty` here, but perhaps it's more correct to specify columns when such are present. What do you think?

File: tck/features/WithAcceptance.feature [hunk 15]
```diff
@@ -0,0 +1,361 @@
...
+      | nestedMap.foo.bar |
+      | 'baz'             |
+    And no side effects
+
+  Scenario: Connected components succeeding WITH
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A)-[:REL]->(x:X)
+      CREATE (b:B)
```
Comment (Mats-SX) [inline tck/features/WithAcceptance.feature]: Could be anonymous

File: tck/features/WithAcceptance.feature [hunk 16]
```diff
@@ -0,0 +1,361 @@
...
+    Then the result should be:
+      | m    | n    | x    |
+      | (:B) | (:A) | (:X) |
+    And no side effects
+
+  Scenario: Single WITH using a predicate and aggregation
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a {prop: 43}), (b {prop: 42})
```
Comment (Mats-SX) [inline tck/features/WithAcceptance.feature]: May be anonymous

**Comment by Mats-SX** [comment]
@petraselmer I cross-referenced to the original, and I find that your amendments are good improvements to the test cases. I've done many similar amendments myself, like adding labels to nodes, since we can not use the same way of asserting that exactly the same node was returned.

File: tck/features/WithAcceptance.feature (hunk 14 above)
Comment (petraselmer) [inline tck/features/WithAcceptance.feature]: I agree this is better. I think we ought to distinguish between nothing being returned (such as  CREATE statement) versus an empty result set being returned. Will cardify so in existing TCK files...

**Pull request merged**